core/src/main/scala/kafka/server/BrokerServer.scala

```diff
@@ -272,7 +273,7 @@ class BrokerServer(
       val networkListeners = new ListenerCollection()
       config.advertisedListeners.foreach { ep =>
         networkListeners.add(new Listener().
-          setHost(ep.host).
+          setHost(if (ep.host == null || ep.host.trim.isEmpty) InetAddress.getLocalHost.getCanonicalHostName else ep.host).
```

Comment:
Could you use `Utils.isBlank(ep.host)`?

Reply:
Good idea.Thank you for your guidance and submit your feedback. :)